### PR TITLE
Allow path in dbapi connect

### PR DESCRIPTION
**Pull request by nevinpuri**
Just as you can pass a `path` argument to the `dbapi.connect()` function (enabling persistency). This will allow similar functionality as the `Session` api already provides, but using the dbapi.

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added `path` parameter to dbapi

### Documentation entry for user-facing changes
- Not sure this requires a documentation entry, but here it is:
`path`: the folder path to use for database data on disk.
- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by Nevin1901** [comment]
Don't merge yet, I just realized this change isn't fully working

**Comment by Nevin1901** [comment]
I've updated all the code. Should work but I haven't tested.

**Comment by auxten** [comment]
Please add tests for your modification.

**Comment by nevinpuri** [comment]
@auxten I have added a test. It all should work now 👍

File: tests/test_dbapi_persistence.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import unittest
+from chdb import dbapi
+
+test_state_dir = ".state_tmp_auxten_dbapi"
+
+class TestDBAPIPersistence(unittest.TestCase):
+    def test_persistence(self):
+        conn = dbapi.connect(path=test_state_dir)
+        cur = con.cursor()
```
Comment (auxten) [inline tests/test_dbapi_persistence.py:11]: conn?

**Comment by auxten** [comment]
Fixing some test issue
organizations-2000000.zip link expired. Just use 
[organizations-500000.zip](https://github.com/chdb-io/chdb/files/14662379/organizations-500000.zip)

**Pull request merged**